### ISSUE TITLE
Ensure all assemblies that ship in NuGet packages have source code and are built here

### ISSUE THREAD

**Issue by mellinoe** [dotnet/corefx#4243]
The following should be an exhaustive list of libraries that have .NET Native-specific assemblies that we need to build from here. This is all of the ".NET Native" project files we have internally which have a corresponding NuGet package. For some of these, we will need to port additional source code on top of the shared CoreCLR source code we already have here.

- [ ] System.Collections
- [ ] System.Diagnostics.StackTrace
- [ ] System.Diagnostics.Tools
- [ ] System.Diagnostics.Tracing
- [ ] System.Dynamic.Runtime
- [ ] System.IO
- [ ] System.Linq.Expressions
- [ ] System.Net.Http
- [ ] System.Net.Http.Rtc
- [ ] System.Net.Primitives
- [ ] System.Net.WebSockets.Client
- [ ] System.Private.DataContractSerialization
- [ ] System.Private.ServiceModel
- [ ] System.Private.Uri
- [ ] System.Reflection.TypeExtensions
- [ ] System.Runtime
- [ ] System.Runtime.Extensions
- [ ] System.Runtime.Handles
- [ ] System.Runtime.WindowsRuntime
- [ ] System.Security.Cryptography.X509Certificates
- [ ] System.ServiceModel.Duplex
- [ ] System.ServiceModel.Http
- [ ] System.ServiceModel.NetTcp
- [ ] System.ServiceModel.Primitives
- [ ] System.ServiceModel.Security
- [ ] System.Threading.Overlapped
- [ ] System.Threading.Tasks

NOTE: Building some of the partial facade assemblies is dependent on #4244 for .NET Native partial facades.

**Comment by davidsh** [comment]
I think System.Net.Sockets and System.Net.NetworkInformation need to be added to this list.

**Comment by mellinoe** [comment]
> I think System.Net.Sockets and System.Net.NetworkInformation need to be added to this list.

Yeah, you're right. This actually only lists the partial facades that we have on the internal side right now. The two you mentioned are full facades (they type forward everything to System.Private.Networking.dll), and we will need to do something for all of the full facades we build internally. We need to do that for all of our CoreCLR facades anyways.

The libraries in the list above are ones that will need to have source code and projects in the repo here, so it doesn't include the full facades.

**Comment by weshaggard** [comment]
Thanks @mellinoe for creating this. I do think it is worth adding the full facades to the list (or perhaps another list) because we need to also do those.

**Comment by mellinoe** [comment]
> or perhaps another list

I guess it depends how we plan on building all of those facades. Internally we build them all at once, so I figured that they would also be addressed "all at once" when we ported the infrastructure out here. But perhaps we will just decide to build them like the partial facades we already have here and address them individually.

**Comment by weshaggard** [comment]
@mellinoe. Currently @ericstj  and I believe we should have individual projects for building them. However we may need to optimize that some if we find that it ends up taking too much time.

**Comment by mellinoe** [comment]
We are basically done with this, the final remaining items are in PR's. That said, I'm going to move this to RTM since we aren't fully finished with it yet. This is just infrastructure improvement and is more related to the packagine work which is on-going and more likely to finish in the RTM timeframe.

**Comment by mellinoe** [comment]
Since System.Diagnostics.Process (v4.0.0.0) is now merged into master, we have everything on GitHub that we ship with packages, at least according to our initial analyses. There may be a couple stragglers as we move the packaging projects out here wholesale.
